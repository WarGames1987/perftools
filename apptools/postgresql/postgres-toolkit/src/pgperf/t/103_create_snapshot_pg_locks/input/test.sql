-- Create
SELECT pgperf.create_snapshot_pg_locks(0);
SELECT count(*) FROM pgperf.snapshot_pg_locks;

SELECT pgperf.create_snapshot_pg_locks(1);
SELECT count(*) FROM pgperf.snapshot_pg_locks;

SELECT pgperf.create_snapshot_pg_locks(2);
SELECT count(*) FROM pgperf.snapshot_pg_locks;

-- Delete
SELECT pgperf.delete_snapshot_pg_locks(1);
SELECT count(*) FROM pgperf.snapshot_pg_locks;

SELECT pgperf.delete_snapshot_pg_locks(2);
SELECT count(*) FROM pgperf.snapshot_pg_locks;

SELECT pgperf.delete_snapshot_pg_locks(0);
SELECT count(*) FROM pgperf.snapshot_pg_locks;
