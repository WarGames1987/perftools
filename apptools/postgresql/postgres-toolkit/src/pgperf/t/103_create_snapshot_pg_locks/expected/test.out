SELECT pgperf.create_snapshot_pg_locks(0);
 create_snapshot_pg_locks 
--------------------------
 t
(1 row)

SELECT count(*) FROM pgperf.snapshot_pg_locks;
 count 
-------
     4
(1 row)

SELECT pgperf.create_snapshot_pg_locks(1);
 create_snapshot_pg_locks 
--------------------------
 t
(1 row)

SELECT count(*) FROM pgperf.snapshot_pg_locks;
 count 
-------
     8
(1 row)

SELECT pgperf.create_snapshot_pg_locks(2);
 create_snapshot_pg_locks 
--------------------------
 t
(1 row)

SELECT count(*) FROM pgperf.snapshot_pg_locks;
 count 
-------
    12
(1 row)

SELECT pgperf.delete_snapshot_pg_locks(1);
 delete_snapshot_pg_locks 
--------------------------
 t
(1 row)

SELECT count(*) FROM pgperf.snapshot_pg_locks;
 count 
-------
     8
(1 row)

SELECT pgperf.delete_snapshot_pg_locks(2);
 delete_snapshot_pg_locks 
--------------------------
 t
(1 row)

SELECT count(*) FROM pgperf.snapshot_pg_locks;
 count 
-------
     4
(1 row)

SELECT pgperf.delete_snapshot_pg_locks(0);
 delete_snapshot_pg_locks 
--------------------------
 t
(1 row)

SELECT count(*) FROM pgperf.snapshot_pg_locks;
 count 
-------
     0
(1 row)

